src/eigensolver/tridiag_solver/kernels.cu

```diff
@@ -171,20 +170,29 @@ DLAF_GPU_ASSEMBLE_RANK1_UPDATE_VECTOR_TILE_ETI(, float);
 DLAF_GPU_ASSEMBLE_RANK1_UPDATE_VECTOR_TILE_ETI(, double);
 
 template <class T>
-T maxElementInColumnTile(const matrix::Tile<const T, Device::GPU>& tile, whip::stream_t stream) {
-  SizeType len = tile.size().rows();
-  const T* arr = tile.ptr();
-
+__global__ void maxElementInColumnTileOnDevice(const T* begin_ptr, const T* end_ptr,
+                                               T* device_max_el_ptr) {
 #ifdef DLAF_WITH_CUDA
   constexpr auto par = ::thrust::cuda::par;
 #elif defined(DLAF_WITH_HIP)
   constexpr auto par = ::thrust::hip::par;
 #endif
-  auto d_max_ptr = thrust::max_element(par.on(stream), arr, arr + len);
-  T max_el;
-  // TODO: this is a performance pessimization, the value is on device
-  whip::memcpy_async(&max_el, d_max_ptr, sizeof(T), whip::memcpy_device_to_host, stream);
-  return max_el;
+
+  // NOTE: This could also use max_element. However, it fails to compile with
+  // HIP, so we use reduce as an alternative which works with both HIP and CUDA.
+  *device_max_el_ptr = thrust::reduce(par, begin_ptr, end_ptr, *begin_ptr, thrust::maximum<T>());
```

Comment:
This was previously using `max_element`. However, it doesn't compile with HIP in a device function (complains about internals trying to "call a host-device function from a host function"). For some reason `reduce` doesn't have this problem, so I'm using that instead.